# Instead of always building libpulse as a shared library, this patch makes libpulse be built as either static or shared depending on the default library type.

--- a/src/meson.build
+++ b/src/meson.build
@@ -198,7 +198,7 @@ if not get_option('client')
   libpulse_dep = dependency('libpulse', required : true)
   libpulse_simple_dep = dependency('libpulse-simple', required : true)
 else
-  libpulsecommon = shared_library('pulsecommon-' + pa_version_major_minor,
+  libpulsecommon = library('pulsecommon-' + pa_version_major_minor,
     libpulsecommon_sources,
     libpulsecommon_headers,
     include_directories : [configinc, topinc],
--- a/src/pulse/meson.build
+++ b/src/pulse/meson.build
@@ -80,7 +80,7 @@ else
   versioning_link_args = []
 endif
 
-libpulse = shared_library('pulse',
+libpulse = library('pulse',
   libpulse_sources,
   libpulse_headers,
   version : libpulse_version,
